emcc.py

```diff
@@ -770,24 +757,20 @@ def filter_emscripten_options(argv):
     os.environ['EMMAKEN_JUST_CONFIGURE_RECURSE'] = '1'
     ret = run_process(cmd, check=False).returncode
     os.environ['EMMAKEN_JUST_CONFIGURE_RECURSE'] = ''
-    if not os.path.exists(target):
-      # note that emcc -c will cause target to have the wrong value here;
-      # but then, we don't care about bitcode outputs anyhow, below, so
-      # skipping returning early is fine
-      return ret
-    if target.endswith('.js'):
-      shutil.copyfile(target, unsuffixed(target))
-      target = unsuffixed(target)
-    if not target.endswith(OBJECT_FILE_ENDINGS):
-      src = open(target).read()
-      full_node = ' '.join(shared.NODE_JS)
-      if os.path.sep not in full_node:
-        full_node = '/usr/bin/' + full_node # TODO: use whereis etc. And how about non-*NIX?
-      open(target, 'w').write('#!' + full_node + '\n' + src) # add shebang
-      try:
-        os.chmod(target, stat.S_IMODE(os.stat(target).st_mode) | stat.S_IXUSR) # make executable
-      except OSError:
-        pass # can fail if e.g. writing the executable to /dev/null
+    if ret == 0:
```

Comment:
```suggestion
    if ret == 0 and os.path.exists(target):
```

Otherwise a lot of projects cannot be configured (tested with the tip-of-tree build of emscripten). See for example:

<details><summary>Details</summary>
<p>

```bash
mkdir zlib
curl -Ls https://zlib.net/zlib-1.2.11.tar.xz | tar xJC zlib --strip-components=1
cd zlib
emconfigure ./configure --prefix=build --static || cat configure.log
# emmake make install
```

```
--------------------
=== ztest141464.c ===
int foo() { return 0; }
===
/home/kleisauke/emsdk/upstream/emscripten/emcc -c -O ztest141464.c
Traceback (most recent call last):
  File "/home/kleisauke/emsdk/upstream/emscripten/emcc.py", line 3956, in <module>
    sys.exit(run(sys.argv))
  File "/home/kleisauke/emsdk/upstream/emscripten/emcc.py", line 763, in run
    shutil.copyfile(target, unsuffixed(target))
  File "/usr/lib64/python3.7/shutil.py", line 120, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'a.out.js'
(exit code 1)
Compiler error reporting is too harsh for ./configure (perhaps remove -Werror).
** ./configure aborting.
```
</p>
</details>

Reply:
Oops thanks!

I wonder why out tests don't catch this